lib/ably/realtime/channel/channel_manager.rb

```diff
@@ -126,17 +126,31 @@ def connect_if_connection_initialized
         connection.connect if connection.initialized?
       end
 
+      def realtime_request_timeout
+        connection.defaults.fetch(:realtime_request_timeout)
+      end
+
       def send_attach_protocol_message
-        send_state_change_protocol_message Ably::Models::ProtocolMessage::ACTION.Attach
+        send_state_change_protocol_message Ably::Models::ProtocolMessage::ACTION.Attach, :suspended # move to suspended
       end
 
       def send_detach_protocol_message
-        send_state_change_protocol_message Ably::Models::ProtocolMessage::ACTION.Detach
+        send_state_change_protocol_message Ably::Models::ProtocolMessage::ACTION.Detach, channel.state # return to current state
       end
 
-      def send_state_change_protocol_message(state)
+      def send_state_change_protocol_message(new_state, state_if_failed)
+        state_at_time_of_request = channel.state
+        failed_timer = EventMachine::Timer.new(realtime_request_timeout) do
+          if channel.state == state_at_time_of_request
+            error = Ably::Models::ErrorInfo.new(code: 90007, message: "Channel #{new_state} operation failed (timed out)")
```

Comment:
I've been using 90000/408 for channel timeouts. But looking at 90007 now, you're right, it's better - I'll change in ably-js.